Parser/+Workhorse/translate_system_configuration.m

```diff
@@ -0,0 +1,145 @@
+%  adcp_system.system_freq[double] - the ADCP system frequency.
+%  adcp_system.model_name[str] - the ADCP model.
+%  adcp_system.xmit_voltage_scale[double] - Transmission voltage scale factors.
+%
+%             MSB derived fields:
+%
+%  adcp_system.beam_config[str] - the beam configuration string.
+%  adcp_system.beam_angle[double] - the physical beam angle of transducers.
+%
+% Example:
+%
+% %A quarter master, upfacing, with hd xdcr, convex, 150khz adcp
+% lsb = ['11001001'];
+% %4beam no demod, 20deg beam angle
+% msb = ['01000001'];
+% [s] = Workhorse.translate_system_configuration(bin2logical([lsb msb]));
+% assert(strcmpi(s.beam_face_config,'up'))
+% assert(s.xdrc_hd_attached)
+% assert(strcmpi(s.sensor_config,'00'))
+% assert(strcmpi(s.beam_pattern,'convex'))
+% assert(s.system_freq==150.00)
+% assert(strcmpi(s.model_name,'Quartermaster'))
+% assert(s.xmit_voltage_scale==592157)
+% assert(strcmpi(s.beam_config,'4-BEAM JANUS CONFIG'))
+% assert(s.beam_angle==20.00)
+%
+%
+% author: hugo.oliveira@utas.edu.au
+%
+
+narginchk(1, 1)
+
+if ~islogical(sysconfig)
+    errormsg('Input should be a logical representing bit configuration')
+elseif numel(sysconfig) ~= 16
+    errormsg('system config should be composed of 16 logical array representing two bytes')
+end
+
+%map the manual table bits to left to right indexing.
+bitmap = containers.Map({0, 1, 2, 3, 4, 5, 6, 7}, {8, 7, 6, 5, 4, 3, 2, 1});
+
+lsb = sysconfig(1:8);
+adcp_system = struct;
+
+if lsb(bitmap(7))
+    adcp_system.beam_face_config = 'up';
+else
+    adcp_system.beam_face_config = 'down';
+end
+
+adcp_system.xdrc_hd_attached = lsb(bitmap(6));
+adcp_system.sensor_config = logical2bin(lsb([bitmap(5), bitmap(4)]));
+
+if lsb(bitmap(3))
+    adcp_system.beam_pattern = 'convex';
+else
+    adcp_system.beam_pattern = 'concave';
+end
+
+model_bits = [bitmap(2), bitmap(1), bitmap(0)];
+model_id = logical2bin(lsb(model_bits));
+
+switch model_id% see readWorkhorseEnsembles.m table.
+    case '000'
+        adcp_system.system_freq = 75;
+        adcp_system.model_name = 'Long Ranger';
+        adcp_system.xmit_voltage_scale = 2092719;
+    case '001'
+        adcp_system.system_freq = 150;
+        adcp_system.model_name = 'Quartermaster';
+        adcp_system.xmit_voltage_scale = 592157;
+    case '010'
+        adcp_system.system_freq = 300;
+        adcp_system.model_name = 'Sentinel or Monitor';
+        adcp_system.xmit_voltage_scale = 592157;
+    case '011'
+        adcp_system.system_freq = 600;
+        adcp_system.model_name = 'Sentinel or Monitor';
+        adcp_system.xmit_voltage_scale = 380667;
+    case '100'
+        adcp_system.system_freq = 1200;
+        adcp_system.model_name = 'Sentinel or Monitor';
+        adcp_system.xmit_voltage_scale = 253765;
+    case '101'
+        adcp_system.system_freq = 2400;
+        adcp_system.model_name = 'DVS';
+        adcp_system.xmit_voltage_scale = 253765;
+    otherwise
```

Comment:
wouldn't this be better to be in a config file? same with a few other things below. This would avoid writing all the case functions

Reply:
these values are pretty much static. We can't actually write nice structs in config files, unless we go to json/toml which is an overkill here.